.github/workflows/test.yml

```diff
@@ -0,0 +1,85 @@
+name: Test
+on:
+  push:
+    branches:
+    - main
+    - feature/datasets
+    - tz/pypi
+  pull_request:
+    branches:
+    - main
+    - feature/datasets
+jobs:
+  test:
+    name: ${{ matrix.lang }} tests on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        lang: [Python]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.9'
+
+    - name: Install Poetry
+      run: |
+        pip install --upgrade pip
+        pip install poetry coveragepy-lcov
+
+    - name: Cache Poetry virtualenv
+      uses: actions/cache@v1
+      id: cache
+      with:
+        path: ~/.virtualenvs
+        key: poetry-${{ hashFiles('**/poetry.lock') }}
+        restore-keys: |
+          poetry-${{ hashFiles('**/poetry.lock') }}
+
+    - name: Set Poetry config
+      run: |
+        poetry config virtualenvs.in-project false
+        poetry config virtualenvs.path ~/.virtualenvs
+
+    - name: Install Dependencies
+      run: |
+        poetry install -E dask -E spark
+        poetry run pip install --upgrade git+https://github.com/zillow/metaflow.git@feature/kfp
+      if: steps.cache.outputs.cache-hit != 'true'
+
+    - name: Code Quality Check
+      run: |
+        poetry run black datasets
```

Comment:
Guessing you don't want `black` during CICD to actually mutate the files, I think you need to run it with a `--check`  to actually validate the formatting and not perform any mutations.